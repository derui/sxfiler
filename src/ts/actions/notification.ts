import { Notification } from "../domains/notification";
import { AppAction } from "./type";

export enum ActionTypes {
  timeout = "notification_timeout",
  notify = "notification_notify",
  remove = "notification_remove",
}

type TimeoutAction = AppAction<ActionTypes.timeout, { notificationId: string }>;
type NotifyAction = AppAction<ActionTypes.notify, { notification: Notification }>;
type RemoveAction = AppAction<ActionTypes.remove, { notificationId: string }>;

export type Actions = TimeoutAction | NotifyAction | RemoveAction;

const timeout = (notificationId: string): TimeoutAction => {
  return { type: ActionTypes.timeout, notificationId };
};

const notify = (notification: Notification): NotifyAction => {
  return { type: ActionTypes.notify, notification };
};

const remove = (notificationId: string): RemoveAction => {
  return { type: ActionTypes.remove, notificationId };
};

export const actions = { timeout, notify, remove };
