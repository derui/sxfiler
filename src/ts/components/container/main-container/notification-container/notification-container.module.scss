@import "../../../ui/commom-style/_colors.scss";
@import "../../../ui/commom-style/_variables.scss";
@import "../../../ui/commom-style/reset.module.css";

.root {
  position: relative;

  height: 100%;
  width: 100%;
  max-width: 100%;

  border-top: 1px solid $sol_base2;
}
