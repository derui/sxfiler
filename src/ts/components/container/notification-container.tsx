import * as React from "react";

interface Prop {}

const NotificationContainer: React.FC<Prop> = props => {
  return null;
};

export default NotificationContainer;
