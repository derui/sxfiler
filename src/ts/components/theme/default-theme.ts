// styled-components.ts
import * as styledComponents from "styled-components";

export interface Theme {
  baseFontSize: string;
  baseBorderRadius: string;
  headerShadow: string;
  boxShadow: string;
  colors: {
    base03: string;
    base02: string;
    base01: string;
    base00: string;
    base0: string;
    base1: string;
    base2: string;
    base3: string;
    yellow: string;
    orange: string;
    red: string;
    magenta: string;
    violet: string;
    blue: string;
    cyan: string;
    green: string;
  };
  spaces: {
    large: string;
    base: string;
    nano: string;
    small: string;
  };
}

export const Theme = {
  baseFontSize: "13.5px",
  baseBorderRadius: "4px",
  headerShadow: "0px 7px 3px rgba(0, 0, 0, 0.2)",
  boxShadow: "4px 4px 7px rgba(0, 0, 0, 0.2), -4px 0px 7px rgba(0, 0, 0, 0.2)",

  colors: {
    /* color schema of solalized-dark (from http://ethanschoonover.com/solarized) */
    base03: "#002b36",
    base02: "#073642",
    base01: "#586e75",
    base00: "#657b83",
    base0: "#839496",
    base1: "#93a1a1",
    base2: "#eee8d5",
    base3: "#fdf6e3",
    yellow: "#b58900",
    orange: "#cb4b16",
    red: "#dc322f",
    magenta: "#d33682",
    violet: "#6c71c4",
    blue: "#268bd2",
    cyan: "#2aa198",
    green: "#859900",
  },
  spaces: {
    large: "16px",
    base: "8px",
    small: "4px",
    nano: "2px",
  },
};

const { default: styled, css, ThemeProvider } = styledComponents as styledComponents.ThemedStyledComponentsModule<
  Theme
>;

export { styled, css, ThemeProvider };
