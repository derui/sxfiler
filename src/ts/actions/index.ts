import { Actions as NotificationActions } from "./notification";
import { Actions as FilerActions } from "./filer";
import { Actions as KeymapActions } from "./key-map";
import { Actions as TaskActions } from "./task";
import { Actions as HistoryActions } from "./history";
import { Actions as FinderAction } from "./finder";

export type Actions = NotificationActions | FilerActions | KeymapActions | TaskActions | HistoryActions | FinderAction;

export { ActionTypes } from "./type";
