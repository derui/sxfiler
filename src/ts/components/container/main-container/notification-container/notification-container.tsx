import * as React from "react";

import * as Element from "../../../ui/element/element";
import { ContextLike } from "../../../../context";
import { State } from "../../../../states/notification";
import { createUseCase } from "../../../../usecases/notification/timeout";
import { Component as NotificationList } from "../../../project/notification-list/notification-list";
import LocatorContext from "../../../../locator";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require("./notification-container.module.scss");

export interface Props {
  state: State;
}

export class Component extends React.PureComponent<Props> {
  public render() {
    const { state } = this.props;
    const timeouts = state.timeouts.messages.map(v => v.id);

    return (
      <LocatorContext.Consumer>
        {({ context }) => {
          if (!context) {
            return null;
          }

          return (
            <Element.Component className={styles.root}>
              <NotificationList
                notifications={state.notifications.messages}
                timeouts={timeouts}
                onNotificationHidden={this.handleNotificationHidden(context)}
              />
            </Element.Component>
          );
        }}
      </LocatorContext.Consumer>
    );
  }

  private handleNotificationHidden = (context: ContextLike) => (id: string) => {
    context.execute(createUseCase(), {
      notificationId: id,
    });
  };
}
