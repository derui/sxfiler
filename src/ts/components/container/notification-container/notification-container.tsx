import * as React from "react";
import { Context } from "../../../context";
import { State } from "../../../types/store-state/notification";
import TimeoutUseCase from "../../../usecases/notification/timeout";
import { TimeoutCallback } from "../../project/notification-item/notification-item";
import * as NotificationList from "../../project/notification-list/notification-list";

// tslint:disable-next-line
const styles = require("./notification-container.module.scss");

export interface Props {
  context: Context;
  state: State;
}

export class Component extends React.PureComponent<Props> {
  public render() {
    const { state } = this.props;

    return (
      <div className={styles.root}>
        <NotificationList.Conmonent
          notifications={state.notifications.items}
          onItemTimeouted={this.handleItemTimeouted}
        />
      </div>
    );
  }
  private handleItemTimeouted: TimeoutCallback = (id: string) => {
    this.props.context.execute(new TimeoutUseCase(), {
      notificationId: id,
    });
  };
}
