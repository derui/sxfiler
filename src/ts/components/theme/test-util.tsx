import * as React from "react";
import { Theme, ThemeProvider } from "./default-theme";

/**
   wrap element with default theme.
 */
export const wrap = (e: React.ReactElement) => {
  return <ThemeProvider theme={Theme}>{e}</ThemeProvider>;
};
