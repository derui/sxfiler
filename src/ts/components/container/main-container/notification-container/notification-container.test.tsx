import * as React from "react";
import renderer from "react-test-renderer";
import { wrap } from "@/components/theme/test-util";

import { Actions } from "@/actions/notification";
import { createNotifications } from "@/domains/progress-notifications";
import { UseCaseLike } from "@/usecases/type";
import { Component } from "./notification-container";
import { LocatorContext } from "@/locator";
import { createProgress } from "@/domains/progress-notification";

const context = (done?: (arg: any) => void) => ({
  use<P>(_: UseCaseLike<Actions, P>) {
    return {
      execute(args: P) {
        if (done) {
          done(args);
        }
      },
    };
  },
});

describe("Container", () => {
  describe("Notification Container", () => {
    it("should render correctly when context not initialized yet", () => {
      const locator = {};
      const state = {
        progresses: createNotifications([
          createProgress("progress", {
            process: "process",
            target: 100,
            current: 10,
          }),
        ]),
        timeouts: createNotifications([]),
      };
      const tree = renderer
        .create(
          wrap(
            <LocatorContext.Provider value={locator}>
              <Component state={state} />
            </LocatorContext.Provider>
          )
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render correctly when context initialized", () => {
      const locator = { context: context(jest.fn()) };
      const state = {
        progresses: createNotifications([
          createProgress("progress", {
            process: "process",
            target: 100,
            current: 10,
          }),
        ]),
        timeouts: createNotifications([]),
      };
      const tree = renderer
        .create(
          wrap(
            <LocatorContext.Provider value={locator}>
              <Component state={state} />
            </LocatorContext.Provider>
          )
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
