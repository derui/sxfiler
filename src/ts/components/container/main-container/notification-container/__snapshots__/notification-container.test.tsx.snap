// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Notification Container should render correctly when context initialized 1`] = `
<div
  className="sc-EHOje lmepRT"
>
  <div
    className="sc-ifAKCX ipbtqc"
    role="list"
  >
    <div
      aria-selected={false}
      className="sc-bdVaJa gneklD"
      role="listitem"
    >
      <span
        className="sc-bwzfXH fCwXYg"
      >
        process
      </span>
      <div
        className="sc-htpNat hkuyqD"
      >
        <div
          className="sc-bxivhb MqTMw"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Container Notification Container should render correctly when context not initialized yet 1`] = `null`;
